nbdime/merging/decisions.py

```diff
@@ -311,10 +311,12 @@ def _sort_key(k):
 """
     ret = []
     for s in k.common_path:
-        s = (s if isinstance(s, (int, text_type)) else s.decode())
-
-        if isinstance(s, text_type) and s.isnumeric() or isinstance(s, int):
-            ret.append(('', -int(s)))
+        if not isinstance(s, (int, text_type)):
+            s = s.decode("utf8")
+        if isinstance(s, text_type) and s.isdecimal():
+            s = int(s)
```

Comment:
Reading a little further, I think what we want here is to use a regex `re.compile(r"^[-+]?\d+$")`. This way we exclude any other special unicode stuff or digit separators that might otherwise be counted.